File: stripe/src/test/java/com/stripe/android/StripeSSLSocketFactoryTest.java
package com.stripe.android;

import org.junit.Test;

import static org.junit.Assert.assertArrayEquals;

public class StripeSSLSocketFactoryTest {

    @Test
    public void getEnabledProtocols() {
        final String[] defaultProtocols = {"protocol"};

        assertArrayEquals(new String[]{"protocol"},
                new StripeSSLSocketFactory(false, false)
                        .getEnabledProtocols(defaultProtocols));
        assertArrayEquals(new String[]{"protocol", "TLSv1.2"},
                new StripeSSLSocketFactory(false, true)
                        .getEnabledProtocols(defaultProtocols));
        assertArrayEquals(new String[]{"protocol", "TLSv1.1", "TLSv1.2"},
                new StripeSSLSocketFactory(true, true)
                        .getEnabledProtocols(defaultProtocols));
    }
}

File: stripe/src/test/java/com/stripe/android/StripeSSLSocketFactoryTest.kt
package com.stripe.android

import com.google.common.truth.Truth.assertThat
import kotlin.test.Test

class StripeSSLSocketFactoryTest {

    @Test
    fun getEnabledProtocols() {
        val defaultProtocols = arrayOf("protocol")

        assertThat(
            StripeSSLSocketFactory(tlsv11Supported = false, tlsv12Supported = false)
                .getEnabledProtocols(defaultProtocols)
        ).isEqualTo(
            arrayOf("protocol")
        )

        assertThat(
            StripeSSLSocketFactory(tlsv11Supported = false, tlsv12Supported = true)
                .getEnabledProtocols(defaultProtocols)
        ).isEqualTo(
            arrayOf("protocol", "TLSv1.2")
        )

        assertThat(
            StripeSSLSocketFactory(tlsv11Supported = true, tlsv12Supported = true)
                .getEnabledProtocols(defaultProtocols)
        ).isEqualTo(
            arrayOf("protocol", "TLSv1.1", "TLSv1.2")
        )
    }
}
